pkg/controllers/vdb/replicatedupgrade_reconciler.go

```diff
@@ -330,6 +336,35 @@ func (r *ReplicatedUpgradeReconciler) sandboxReplicaGroupB(ctx context.Context)
 	return ctrl.Result{}, nil
 }
 
+// postPromoteSubclustersInSandboxMsg will update the status message to indicate that
+// we are going to prmote subclusters in sandbox.
+func (r *ReplicatedUpgradeReconciler) postPromoteSubclustersInSandboxMsg(ctx context.Context) (ctrl.Result, error) {
+	return r.postNextStatusMsg(ctx, promoteSubclustersInSandboxInx)
+}
+
+// promoteReplicaBSubclusters promotes all of the secondaries in replica group B whose
+// parent subcluster is primary
+func (r *ReplicatedUpgradeReconciler) promoteReplicaBSubclusters(ctx context.Context) (ctrl.Result, error) {
+	sb := r.VDB.GetSandboxStatus(r.sandboxName)
+	if sb == nil {
+		return ctrl.Result{}, fmt.Errorf("could not find sandbox %q", r.sandboxName)
```

Comment:
Should we return "ctrl.Result{Requeue: true}, nil" here because the sandbox could be in progress?